specs/src/specs/data_structures.md

```diff
@@ -0,0 +1,994 @@
+- [Erasure Coding](#erasure-coding)
+  - [Reed-Solomon Erasure Coding](#reed-solomon-erasure-coding)
+  - [2D Reed-Solomon Encoding Scheme](#2d-reed-solomon-encoding-scheme)
+  - [Share](#share)
+  - [Arranging Available Data Into Shares](#arranging-available-data-into-shares)
+- [Available Data](#available-data)
+  - [TransactionData](#transactiondata)
+    - [WrappedTransaction](#wrappedtransaction)
+    - [Transaction](#transaction)
+    - [SignedTransactionData](#signedtransactiondata)
+      - [SignedTransactionDataTransfer](#signedtransactiondatatransfer)
+      - [SignedTransactionDataMsgPayForData](#signedtransactiondatamsgpayfordata)
+      - [SignedTransactionDataCreateValidator](#signedtransactiondatacreatevalidator)
+      - [SignedTransactionDataBeginUnbondingValidator](#signedtransactiondatabeginunbondingvalidator)
+      - [SignedTransactionDataUnbondValidator](#signedtransactiondataunbondvalidator)
+      - [SignedTransactionDataCreateDelegation](#signedtransactiondatacreatedelegation)
+      - [SignedTransactionDataBeginUnbondingDelegation](#signedtransactiondatabeginunbondingdelegation)
+      - [SignedTransactionDataUnbondDelegation](#signedtransactiondataunbonddelegation)
+      - [SignedTransactionDataBurn](#signedtransactiondataburn)
+      - [SignedTransactionRedelegateCommission](#signedtransactionredelegatecommission)
+      - [SignedTransactionRedelegateReward](#signedtransactionredelegatereward)
+  - [IntermediateStateRootData](#intermediatestaterootdata)
+    - [WrappedIntermediateStateRoot](#wrappedintermediatestateroot)
+    - [IntermediateStateRoot](#intermediatestateroot)
+  - [EvidenceData](#evidencedata)
+    - [Evidence](#evidence)
+    - [PublicKey](#publickey)
+    - [Vote](#vote)
+  - [MessageData](#messagedata)
+    - [Message](#message)
+- [State](#state)
+  - [StateElement](#stateelement)
+  - [Account](#account)
+  - [Delegation](#delegation)
+  - [Validator](#validator)
+  - [ActiveValidatorCount](#activevalidatorcount)
+  - [ActiveVotingPower](#activevotingpower)
+  - [ProposerBlockReward](#proposerblockreward)
+  - [ProposerInitialVotingPower](#proposerinitialvotingpower)
+  - [ValidatorQueueHead](#validatorqueuehead)
+  - [PeriodEntry](#periodentry)
+  - [Decimal](#decimal)
+  - [MessagePaid](#messagepaid)
+  - [MessagePaidHead](#messagepaidhead)
+- [Consensus Parameters](#consensus-parameters)
+
+## Data Structures Overview
+
+![fig: Block data structures.](./figures/block_data_structures.svg)
+
+## Type Aliases
+
+| name                        | type                        |
+|-----------------------------|-----------------------------|
+| [`Address`](#address)       | `byte[32]`                  |
+| `Amount`                    | `uint64`                    |
+| `Graffiti`                  | `byte[MAX_GRAFFITI_BYTES]`  |
+| [`HashDigest`](#hashdigest) | `byte[32]`                  |
+| `Height`                    | `int64`                     |
```

Comment:
[no change needed] I'm not sure why height is a signed integer rather than unsigned but this conforms to tendermint/tendermint ([ref](https://github.com/tendermint/tendermint/blob/aa303edaefeb8a0856d7494cca24e51f6f3e2393/state/errors.go#L13)) so no change needed

Reply:
(Invalid) reasoning here: https://blog.cosmos.network/choosing-a-type-for-blockchain-height-beware-of-unsigned-integers-714804dddf1d